front/src/app/navigation/map-page/map-page.component.ts

```diff
@@ -82,30 +87,30 @@ export class MapPageComponent implements OnInit, OnDestroy, AfterViewInit {
     return leafletBody;
   }
 
-  async ngAfterViewInit() {
-    this.locationDrawer.SetState(DrawerState.Hidden);
-    this.routeDrawer.SetState(DrawerState.Hidden);
-    this.inNavigationDrawer.SetState(DrawerState.Hidden);
-    await this.searchDrawer.SetState(IonicBottomDrawerComponent.GetRecommendedDrawerStateForDevice());
-  }
-
   async ngOnInit() {
-    if (this.model.recentSearches === null || this.model.recentSearches.length === 0) {
+    if (!this.model.recentSearches || this.model.recentSearches.length === 0) {
       const searches = await SearchResultProvider.readRecentSearch(this.storage);
       if (searches !== null) {
         this.model.recentSearches = searches;
         console.log(this.model.recentSearches);
       }
     }
 
-    if (this.model.availableCampus.length === 0)
-    {
+    if (!this.model.availableCampus || this.model.availableCampus.length === 0) {
       this.model.availableCampus = await this.dataService.get_campus_search().toPromise()
     }
 
-    for (const campus of this.model.availableCampus) {
-      this.availableCampus.push(new CampusViewModel(campus))
+    if (this.model.availableCampus) {
+      for (const campus of this.model.availableCampus) {
+        this.availableCampus.push(new CampusViewModel(campus))
+      }
     }
+
+    await Promise.all([
+        this.searchDrawer.SetState(IonicBottomDrawerComponent.GetRecommendedDrawerStateForDevice()),
+        this.locationDrawer.SetState(DrawerState.Hidden),
```

Comment:
funktioniert das sicher im `onInit`? Ich dachte der muss die HTML Elemente initalisiert haben, damit man das so aufrufen kann.

Reply:
ionViewWillEnter kommt laut ionic direkt nach ngOnInit

`The former [ionViewWillEnter] fires right after ngOnInit`